### PR TITLE
TemporaryFile(): Handle one more case correctly

### PR DESCRIPTION
**Pull request by srittau**
Also add tests.

### CODE REVIEW

**Comment by srittau** [comment]
I verified the test types against runtime.

File: stdlib/tempfile.pyi [hunk 1]
```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```
Comment (AlexWaygood) [inline stdlib/tempfile.pyi:1]: You added this one to the wrong file ;)

File: test_cases/stdlib/check_tempfile.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from __future__ import annotations
+
+import io
+import sys
+from tempfile import TemporaryFile
+from typing_extensions import assert_type
+
+if sys.platform != "win32":
```
[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply test_cases/stdlib/check_tempfile.py:8]: TemporaryFile works very differently on Windows

[the remark this answers is not in the document]
Comment (AlexWaygood) [reply test_cases/stdlib/check_tempfile.py:8]: Correction: Temporary

[the remark this answers is not in the document]
Comment (Akuli) [reply test_cases/stdlib/check_tempfile.py:8]: I didn't notice that the stub has a similar if statement.

**Comment by srittau** [comment]
Could I ping this again? Seems like a simple enough fix and is a prerequisite for deciding what to do with #10229.

**Pull request merged**